### ISSUE TITLE
Screen Reader: Create Expenses: The amount field has no role announced

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77570]
## Action Performed:
Precondition: The User should be logged in

1. Using iOS VoiceOver or Android TalkBack, open the Expensify app
2. Swipe to the Create '+' button and double-tap to activate
3. Swipe to the 'Create Expense' button and double-tap to activate
4. Swipe to the 'Amount' input field
5. Note how the control is announced

## Expected Result:
The control should be announced as a text field or edit field.

## Actual Result:
When a screen reader user focuses on the amount field, it announces only the placeholder, but omits the role ("text field" or "edit box"). This ambiguity leaves the user unsure whether the element is a static display or an editable field.

## Area issue was found in:
Create Expenses

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure the text field is correctly implemented with the edit field role.
https://reactnative.dev/docs/touchablewithoutfeedback#accessibilityrole

## Environment:
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

https://github.com/user-attachments/assets/4650e898-4570-43f8-820e-67e73247f37e

<img width="828" height="1792" alt="Image" src="https://github.com/user-attachments/assets/49dc3aea-91ef-4931-a69f-c411513bdee9" />

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7006083
**Issue reported by:** NA
**Slack conversation:** NA
